lib/Conversion/SeqToSV/FirRegLowering.cpp

```diff
@@ -51,6 +51,144 @@ getForwardSliceSimple(Operation *root,
   }
 }
 
+// Construct SCC.
+namespace circt {
+struct ValueSCC {
+  llvm::DenseMap<Value, size_t> order;
+  llvm::DenseMap<Value, size_t> lowlink;
+  llvm::DenseMap<Value, size_t> componentId;
```

Comment:
Can we create a helper struct `struct Node {order;lowlink;componentId}` and combine these three maps?